rpc/core/tx.go

```diff
@@ -63,52 +61,38 @@ func (env *Environment) TxSearch(
 		return nil, ErrQueryLength{len(query), maxQueryLength}
 	}
 
-	q, err := cmtquery.New(query)
-	if err != nil {
-		return nil, err
+	// if orderBy is not "asc", "desc", or blank, return error
+	if orderBy != "" && orderBy != "asc" && orderBy != "desc" {
```

Comment:
I think this condition is incorrect.

```suggestion
	switch orderBy {
	    case "", "asc", "desc":
	    default:
	    
```

Reply:
I am not seeing an issue with the condition

Reply:
This logic branch confirms that it is either blank, "asc", or "desc". We then set OrderDesc here

https://github.com/osmosis-labs/cometbft/blob/d4ed02ce870faa8b5a76f163d7026920ad18c75d/rpc/core/tx.go#L82-L87

Which is equal to the previous logic (if "desc", its in descending order, if "" or "asc", its in ascending order)

Reply:
```
// if orderBy is not "asc", "desc", or blank, return error
	if orderBy != "" && orderBy != "asc" && orderBy != "desc" {
	```
	
**You say if OR OR
But your code is if AND AND**

Reply:
Im still not seeing the problem 😥

We want to check if the value is any one of "", "asc", or "desc". 

If the value was "bar", the result would be true true true, causing the error

If the value is "", the result would be false true true, continuing on with the method, etc etc.

Reply:
Right, sorry for confusion.